GameModel.cpp

```diff
@@ -231,6 +233,35 @@ namespace gameModel{
         return false;
     }
 
+    auto Environment::getTeam(const Player &player) const -> const Team& {
+        if (this->team1.hasMember(player)) {
+            return this->team1;
+        }
+        else if (this->team2.hasMember(player)) {
+            return this->team2;
+        }
+
+        throw std::runtime_error("The Player isn't part of any team.");
+    }
+
+    void Environment::placePlayerOnRandomFreeCell(Player &player) {
+        Position pos;
+        if (this->team1.hasMember(player)) {
+            do {
+                pos = Position(gameController::rng(3,7), gameController::rng(0,12));
+            }
+            while (getCell(pos) != gameModel::Cell::Standard || getPlayer(pos).has_value());
```

Comment:
Geht das nicht schöner? Was wenn kein Feld frei?

Reply:
Kein Feld frei ist bei der geg. Spielerzahl unmöglich.
und so genau weiß ich leider nicht wie ich das zufällig Stellen auf eine freie Zelle in der eigenen hälft optimieren soll, weil wenn die zufällig generierte Zelle nicht frei ist muss eine andere gesucht werden.

Reply:
Du könntest zum Beispiel eine Liste mit möglichen Feldern nehmen und darin einen zufälligen Index bestimmen, das terminiert garantiert